crates/sui-transaction-builder/src/object.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use sui_types::types::ObjectDigest;
+use sui_types::types::ObjectId;
+use sui_types::types::ObjectReference;
+
+use crate::Error;
+
+/// Type representing potentially unresolved object types, with a builder API.
+#[derive(Clone, Debug)]
+pub struct Object {
```

Comment:
What value does this provide that isn't already provided by `UnresolvedInputArgument`?

Reply:
Yeah, I don't think we need this anymore.

Reply:
Would it be ok to add some constructors to `UnresolvedInputArgument`. I think it's a bit nicer to do
```rust
let coin_to_merge = tx.input(Object::owned(
            coin1_obj.object_id(),
            coin1_version,
            coin1_digest,
        ));
```
rather than
```
        let a = tx.input(UnresolvedInputArgument {
            kind: Some(UnresolvedInputArgumentKind::ImmutableOrOwned),
            object_id: Some(coin1),
            version: Some(coin1_version),
            digest: Some(coin1_digest),
            mutable: None,
            value: None,
        });

```

Reply:
Oh certainly, let's do that!

Reply:
How about `initial_shared_version` in `UnresovledInputArgument` if it is a `Shared` kind of object?

Reply:
I see that `version` has an alias `initial_shared_version`. That's a bit confusing to me 😅